examples/morpheusvm/tests/e2e/e2e_test.go

```diff
@@ -46,22 +54,91 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 	expectedABI, err := abi.NewABI(vm.ActionParser.GetRegisteredTypes(), vm.OutputParser.GetRegisteredTypes())
 	require.NoError(err)
 
-	// Import HyperSDK e2e test coverage and inject MorpheusVM name
-	// and workload factory to orchestrate the test.
-	spamHelper := throughput.SpamHelper{
-		KeyType: auth.ED25519Key,
-	}
-
 	authFactories := testingNetworkConfig.AuthFactories()
-	spamKey := authFactories[0]
 	generator := workload.NewTxGenerator(authFactories[1])
-	tc := e2e.NewTestContext()
-	he2e.SetWorkload(testingNetworkConfig, generator, expectedABI, &spamHelper, spamKey)
 
-	return fixture.NewTestEnvironment(tc, flagVars, owner, testingNetworkConfig, consts.ID).Marshal()
+	registry := prometheus.NewRegistry()
+	tracker, err := hload.NewPrometheusTracker[ids.ID](registry)
+	require.NoError(err)
+
+	he2e.SetWorkload(
+		testingNetworkConfig,
+		generator,
+		expectedABI,
+		loadTxGenerators,
+		tracker,
+		hload.ShortBurstOrchestratorConfig{
+			TxsPerIssuer: 1_000,
+			Timeout:      20 * time.Second,
+		},
+		hload.DefaultGradualLoadOrchestratorConfig(),
+		createTransfer,
+	)
+
+	env := fixture.NewTestEnvironment(
+		e2e.NewTestContext(),
+		flagVars,
+		owner,
+		testingNetworkConfig,
+		consts.ID,
+	)
+
+	cleanUpFunc, err := he2e.ExposeMetrics(context.Background(), env, registry)
+	require.NoError(err)
+
+	ginkgo.DeferCleanup(func() {
+		require.NoError(cleanUpFunc())
+	})
+
+	return env.Marshal()
 }, func(envBytes []byte) {
 	// Run in every ginkgo process
 
 	// Initialize the local test environment from the global state
 	e2e.InitSharedTestEnvironment(ginkgo.GinkgoT(), envBytes)
 })
+
+func createTransfer(to codec.Address, amount uint64, nonce uint64) chain.Action {
+	return &actions.Transfer{
+		To:    to,
+		Value: amount,
+		Memo:  binary.BigEndian.AppendUint64(nil, nonce),
+	}
+}
+
+func loadTxGenerators(
+	ctx context.Context,
+	uri string,
+	authFactories []chain.AuthFactory,
+) ([]hload.TxGenerator[*chain.Transaction], error) {
+	lcli := vm.NewJSONRPCClient(uri)
+	ruleFactory, err := lcli.GetRuleFactory(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	numFactories := len(authFactories)
+	balances := make([]uint64, numFactories)
+	// Get balances
+	for i, factory := range authFactories {
+		balance, err := lcli.Balance(ctx, factory.Address())
+		if err != nil {
+			return nil, err
+		}
+		balances[i] = balance
+	}
+
+	cli := jsonrpc.NewJSONRPCClient(uri)
+	unitPrices, err := cli.UnitPrices(ctx, false)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create tx generator
+	txGenerators := make([]hload.TxGenerator[*chain.Transaction], numFactories)
+	for i := 0; i < numFactories; i++ {
+		txGenerators[i] = load.NewTxGenerator(authFactories[i], ruleFactory, balances[i], unitPrices)
+	}
+
+	return txGenerators, nil
+}
```

Comment:
(not necessary for this PR) - I think we'll end up moving this next to `load.NewTxGenerator` in the future to re-use outside of e2e tests as @Elvis339 has requested